processor/src/main/java/org/mapstruct/ap/internal/model/source/selector/XmlElementDeclSelector.java

```diff
@@ -112,4 +119,38 @@ else if ( nameMatches.size() > 0 ) {
             return methods;
         }
     }
+
+    private Pair getMatch(SourceMethod sourceMappingMethod, final String targetPropertyName) {
+        final TypeMirror startingMirror = sourceMappingMethod.getExecutable().getReturnType();
+        TypeMirror currentMirror = startingMirror;
+        TypeElement currentElement = sourceMappingMethod.getResultType().getTypeElement();
+
+        while (currentElement != null) {
+            for (Element enclosed : currentElement.getEnclosedElements()) {
+                if (enclosed.getKind().equals( ElementKind.FIELD ) && targetPropertyName != null
```

Comment:
yes it may be null. If I remember correctly, I broke like 99% of the unittests at first because I did not expect SelectionCriteria.getTargetPropertyName() (which is passed into "targetPropertyName") to be null :)